sklearn/preprocessing/_encoders.py

```diff
@@ -692,7 +692,9 @@ def get_feature_names(self, input_features=None):
 
         feature_names = []
         for i in range(len(cats)):
-            names = [input_features[i] + "_" + str(t) for t in cats[i]]
+            if type(input_features) is not str:
```

Comment:
input_features is an array or a list so `type(input_features)` will be `list` or `numpy.ndarray` which is not what you intended.

I think you meant:

```suggestion
            if not isinstance(input_features[i], str):
```